src/test/java/teammates/test/BaseTestCaseWithSqlDatabaseAccess.java

```diff
@@ -47,20 +65,113 @@ protected void verifyEquals(BaseEntity expected, ApiOutput actual) {
+            assertEquals(expectedFeedbackSession.getGracePeriod().toMinutes(),
+                    actualFeedbackSession.getGracePeriod().longValue());
+            assertEquals(expectedFeedbackSession.isClosingEmailEnabled(),
+                    actualFeedbackSession.getIsClosingEmailEnabled());
+            assertEquals(expectedFeedbackSession.isPublishedEmailEnabled(),
+                    actualFeedbackSession.getIsPublishedEmailEnabled());
+        } else if (expected instanceof Instructor) {
+            Instructor expectedInstructor = (Instructor) expected;
+            InstructorData actualInstructor = (InstructorData) actual;
+            assertEquals(expectedInstructor.getCourseId(), actualInstructor.getCourseId());
+            assertEquals(expectedInstructor.getName(), actualInstructor.getName());
+            assertEquals(expectedInstructor.getEmail(), actualInstructor.getEmail());
+            assertEquals(expectedInstructor.getRegKey(), actualInstructor.getKey());
+            assertEquals(expectedInstructor.isDisplayedToStudents(), actualInstructor.getIsDisplayedToStudents());
+            assertEquals(expectedInstructor.getDisplayName(), actualInstructor.getDisplayedToStudentsAs());
+            assertEquals(expectedInstructor.getRole(), actualInstructor.getRole());
+        } else if (expected instanceof Notification) {
+            Notification expectedNotification = (Notification) expected;
+            NotificationData actualNotification = (NotificationData) actual;
+            assertEquals(expectedNotification.getStartTime().toEpochMilli(), actualNotification.getStartTimestamp());
+            assertEquals(expectedNotification.getEndTime().toEpochMilli(), actualNotification.getEndTimestamp());
+            assertEquals(expectedNotification.getStyle(), actualNotification.getStyle());
+            assertEquals(expectedNotification.getTargetUser(), actualNotification.getTargetUser());
+            assertEquals(expectedNotification.getTitle(), actualNotification.getTitle());
+            assertEquals(expectedNotification.getMessage(), actualNotification.getMessage());
+            assertEquals(expectedNotification.isShown(), actualNotification.isShown());
+        } else if (expected instanceof Student) {
+            Student expectedStudent = (Student) expected;
+            StudentData actualStudent = (StudentData) actual;
+            assertEquals(expectedStudent.getCourseId(), actualStudent.getCourseId());
+            assertEquals(expectedStudent.getName(), actualStudent.getName());
+            assertEquals(expectedStudent.getEmail(), actualStudent.getEmail());
+            assertEquals(expectedStudent.getRegKey(), actualStudent.getKey());
+            assertEquals(expectedStudent.getComments(), actualStudent.getComments());
+            assertEquals(expectedStudent.getTeamName(), actualStudent.getTeamName());
+            assertEquals(expectedStudent.getSectionName(), actualStudent.getSectionName());
```

Comment:
for getTeamName and getSectionName, there might be null pointer exceptions as a student might not have a team or section, let's comment these out and fix this bug in another PR

Reply:
ah that makes sense. thank you!